distributed/client.py

```diff
@@ -343,7 +343,7 @@ class Client(object):
     distributed.scheduler.Scheduler: Internal scheduler
     """
     def __init__(self, address=None, start=True, loop=None, timeout=3,
-                 set_as_default=True, scheduler_file=None):
+                 set_as_default=True, scheduler_file=None, connection_kwargs=None):
```

Comment:
Ah I see, that does make more sense.  I guess the question then becomes about the following tradeoff

1.  The inconvenience of passing around `connection_kwargs` and the long term maintenance costs of this.
2.  The likelihood that we will want to have different connection kwargs within the same process.  

@pitrou seems to have taken on the maintenance of distributed/comm, so he probably has the most say here, especially on point 1.  

On point number two I don't know of anyone who would do this today, but it doesn't seem that unlikely.  People today are already building some strange things.